src/Collateral.hs

```diff
@@ -80,17 +82,15 @@ mkValidator contractInfo@ContractInfo{..} dat interestPayDate ctx = validate
     validateDebtAmnt :: TxOut -> Bool
     validateDebtAmnt txo = getLoanAmnt (txOutValue txo) >= loanamnt dat
 
-    interestPercentage :: Integer
-    interestPercentage = case (lendDate dat + repayinterval dat) < interestPayDate of
-      True  -> 100
-      False -> (getPOSIXTime loanHeld `multiplyInteger` 100) `divideInteger` getPOSIXTime (repayinterval dat)
-       where
-        loanHeld = interestPayDate - lendDate dat
```

Comment:
I suggest to keep the alias `loanHeld = interestPayDate - lendDate dat`, it helps readability.